HerPublicWebsite.BusinessLogic/Extensions/StringExtensions.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Text.RegularExpressions;
+
+namespace HerPublicWebsite.BusinessLogic.Extensions;
+
+public static class StringExtensions
+{
+    private static Regex[] validPostCodeRegexes = new[]
+    {
+        new Regex(@"^[A-Z]\d \d[A-Z][A-Z]$", RegexOptions.Compiled),          //   AN NAA
+        new Regex(@"^[A-Z]\d\d \d[A-Z][A-Z]$", RegexOptions.Compiled),        //  ANN NAA
+        new Regex(@"^[A-Z][A-Z]\d \d[A-Z][A-Z]$", RegexOptions.Compiled),     //  AAN NAA
+        new Regex(@"^[A-Z][A-Z]\d\d \d[A-Z][A-Z]$", RegexOptions.Compiled),   // AANN NAA
+        new Regex(@"^[A-Z]\d[A-Z] \d[A-Z][A-Z]$", RegexOptions.Compiled),     //  ANA NAA
+        new Regex(@"^[A-Z][A-Z]\d[A-Z] \d[A-Z][A-Z]$", RegexOptions.Compiled) // AANA NAA
```

Comment:
Wouldn't it be better to use a single Regex for this? By my count, you could have:
```regex
^[A-Z]{1,2}(\d|\d\d|\d[A-Z]) \d[A-Z]{2}$
```

Reply:
I find Regexes get complicated really quickly so wanted to keep it as simple as possible so it's easy to map a regex to an allowed format. Using [A-Z]{2} does seem clearer though.